TombEngine/Objects/TR1/tr1_objects.cpp

```diff
@@ -188,6 +189,15 @@ static void StartTrap(ObjectInfo* obj)
 		obj->shadowType = ShadowMode::All;
 		obj->SetupHitEffect(true);
 	}
+
+	obj = &Objects[ID_TEETH_SPIKE_DOOR];
+	if (obj->loaded)
+	{
+		obj->initialise = InitialiseTeethSpikeDoor;
+		obj->control = ControlTeethSpikeDoor;
+		obj->collision = GenericSphereBoxCollision;
+		obj->SetupHitEffect(true);
```

Comment:
Remember me to check for this. This genericSphereBoxCollision uses the ItemFlags[3] to read Damage and apply it to Lara, so right now, this system is activating doDamage 2 times. Once with this collision function that applies 0 damage and other in control function that applies the 400.

There may be other functions more suitable for collisions in the other traps  need to check. Or it could adapt the damage in the itemFlags to avoid the duplicated doDamage call.

Reply:
hmm there are no other functions for trap collision as far as I saw.

Reply:
Resolved on Discord chat.